### PR TITLE
Add innerloop tag for pri0 jobs

### PR DESCRIPTION
**Pull request by jashook**
Everything that is not innerloop will run priority 1.

### CODE REVIEW

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
This is ready to go now.

/cc @dotnet/jit-contrib

**Comment by jashook** [comment]
Summary of changes

```
PR Triggered Jobs

-- Added Jobs --
CentOS7.1 x64 Checked Innerloop Build and Test (checked_centos7.1_innerloop_flow_prtest)
CentOS7.1 x64 Debug Innerloop Build (debug_centos7.1_innerloop_prtest)
OSX10.12 x64 Checked Innerloop Build and Test.* (checked_osx10.12_innerloop_flow_prtest)
Tizen armel Cross Checked Innerloop Build and Test (armel_cross_checked_tizen_innerloop_prtest)
Ubuntu arm Cross Debug Innerloop Build (arm_cross_debug_ubuntu_innerloop_prtest)
Ubuntu arm64 Cross Debug Innerloop Build (arm64_debug_small_page_size_innerloop_prtest)
Ubuntu x64 Checked Innerloop Build and Test.* (checked_ubuntu_innerloop_flow_prtest)
Ubuntu16.04 arm Cross Debug Innerloop Build (arm_cross_debug_ubuntu16.04_innerloop_prtest)
Windows_NT arm Cross Checked Innerloop Build and Test (arm_cross_checked_windows_nt_innerloop_flow_prtest)
Windows_NT arm64 Cross Checked Innerloop Build and Test (arm64_cross_checked_windows_nt_innerloop_flow_prtest)
Windows_NT armlb Cross Checked Innerloop Build and Test (armlb_cross_checked_windows_nt_innerloop_flow_prtest)
Windows_NT x64 Checked Innerloop Build and Test (checked_windows_nt_innerloop_prtest)
Windows_NT x86 Checked Innerloop Build and Test (x86_checked_windows_nt_innerloop_prtest)

-- Removed Jobs --
CentOS7.1 x64 Checked Innerloop Build and Test (checked_centos7.1_flow_prtest)
CentOS7.1 x64 Debug Innerloop Build (debug_centos7.1_prtest)
OSX10.12 x64 Checked Innerloop Build and Test (checked_osx10.12_flow_prtest)
Ubuntu arm64 Cross Debug Innerloop Build (arm64_debug_small_page_size_prtest)
Ubuntu x64 Checked Innerloop Build and Test (checked_ubuntu_flow_prtest)
Windows_NT arm Cross Checked Innerloop Build and Test (arm_cross_checked_windows_nt_flow_prtest)
Windows_NT arm64 Cross Checked Innerloop Build and Test (arm64_cross_checked_windows_nt_flow_prtest)
Windows_NT armlb Cross Checked Innerloop Build and Test (armlb_cross_checked_windows_nt_flow_prtest)
Windows_NT x64 Checked Innerloop Build and Test (checked_windows_nt_prtest)
Windows_NT x86 Checked Innerloop Build and Test (x86_checked_windows_nt_prtest)

-- Changed Jobs --
Ubuntu x64 Innerloop Formatting -> Ubuntu x64 Formatting
Windows_NT x64 Innerloop Formatting -> Windows_NT x64 Formatting

Push Triggered Jobs

-- Added Jobs --
arm_cross_checked_ubuntu16.04_innerloop
arm_cross_checked_ubuntu_innerloop
arm_cross_checked_windows_nt_innerloop
arm_cross_checked_windows_nt_innerloop_flow
arm_cross_debug_ubuntu16.04_innerloop
arm_cross_debug_ubuntu_innerloop
arm_cross_debug_windows_nt_innerloop
arm_cross_debug_windows_nt_innerloop_flow
arm_cross_release_ubuntu16.04_innerloop
arm_cross_release_ubuntu_innerloop
arm_cross_release_windows_nt_innerloop
arm_cross_release_windows_nt_innerloop_flow
armel_cross_checked_tizen_innerloop
armel_cross_debug_tizen_innerloop
armel_cross_release_tizen_innerloop
armlb_cross_checked_windows_nt_innerloop
armlb_cross_checked_windows_nt_innerloop_flow
armlb_cross_debug_windows_nt_innerloop
armlb_cross_debug_windows_nt_innerloop_flow
armlb_cross_release_windows_nt_innerloop
armlb_cross_release_windows_nt_innerloop_flow
checked_centos7.1_innerloop_flow
checked_fedora24_innerloop
checked_osx10.12_innerloop_flow
checked_ubuntu16.04_innerloop
checked_ubuntu16.10_innerloop
checked_ubuntu_innerloop_flow
checked_windows_nt_innerloop
debug_fedora24_innerloop
debug_osx10.12_innerloop_flow
debug_ubuntu16.04_innerloop
debug_ubuntu16.10_innerloop
debug_ubuntu_innerloop_flow
debug_windows_nt_innerloop
release_centos7.1_innerloop_flow
release_fedora24_innerloop
release_osx10.12_innerloop_flow
release_ubuntu16.04_innerloop
release_ubuntu16.10_innerloop
release_ubuntu_innerloop_flow
release_windows_nt_innerloop
x64_arm64_altjit_checked_windows_nt_innerloop
x64_arm64_altjit_debug_windows_nt_innerloop
x64_arm64_altjit_release_windows_nt_innerloop
x86_arm_altjit_checked_windows_nt_innerloop
x86_arm_altjit_debug_windows_nt_innerloop
x86_arm_altjit_release_windows_nt_innerloop
x86_checked_windows_nt_innerloop
x86_debug_windows_nt_innerloop
x86_release_windows_nt_innerloop

-- Removed Jobs --

-- Changed Jobs --

Cron Jobs

-- Added Jobs --
arm64_checked_small_page_size_innerloop (H H/4 * * *)
arm64_checked_small_page_size_innerloop_flow (H H/4 * * *)
arm64_cross_checked_windows_nt_innerloop (H H/4 * * *)
arm64_cross_checked_windows_nt_innerloop_flow (H H/4 * * *)
arm64_cross_debug_windows_nt_innerloop (H H/4 * * *)
arm64_cross_debug_windows_nt_innerloop_flow (H H/4 * * *)
arm64_cross_release_windows_nt_innerloop (H H/4 * * *)
arm64_cross_release_windows_nt_innerloop_flow (H H/4 * * *)
arm64_debug_small_page_size_innerloop (H H/4 * * *)
arm64_debug_small_page_size_innerloop_flow (H H/4 * * *)
arm64_release_small_page_size_innerloop (H H/4 * * *)
arm64_release_small_page_size_innerloop_flow (H H/4 * * *)
x86_checked_ubuntu_innerloop_flow (@daily)
x86_debug_ubuntu_innerloop_flow (@daily)
x86_release_ubuntu_innerloop_flow (@daily)

-- Removed Jobs --

-- Changed Jobs --

Unrefferenced Jobs

-- Added Jobs --
arm64_checked_small_page_size_flow_prtest
arm64_debug_small_page_size_flow_prtest
arm64_release_small_page_size_flow_prtest
checked_debian8.4_innerloop
checked_rhel7.2_innerloop
debug_centos7.1_innerloop
debug_centos7.1_prtest
debug_debian8.4_innerloop
debug_osx10.12_innerloop_flow_prtest
debug_rhel7.2_innerloop
debug_ubuntu_innerloop_flow_prtest
debug_windows_nt_innerloop_prtest
release_centos7.1_innerloop_flow_prtest
release_osx10.12_innerloop_flow_prtest
release_ubuntu_innerloop_flow_prtest
release_windows_nt_innerloop_prtest
x64_arm64_altjit_checked_windows_nt_innerloop_bld
x64_arm64_altjit_checked_windows_nt_innerloop_bld_prtest
x64_arm64_altjit_debug_windows_nt_innerloop_bld
x64_arm64_altjit_debug_windows_nt_innerloop_bld_prtest
x64_arm64_altjit_release_windows_nt_innerloop_bld
x64_arm64_altjit_release_windows_nt_innerloop_bld_prtest
x64_checked_windows_nt_gc_reliability_framework_bld
x64_checked_windows_nt_gc_reliability_framework_bld_prtest
x64_checked_windows_nt_standalone_gc_bld
x64_checked_windows_nt_standalone_gc_bld_prtest
x64_release_windows_nt_gc_reliability_framework_bld
x64_release_windows_nt_gc_reliability_framework_bld_prtest
x64_release_windows_nt_ilrt_bld
x64_release_windows_nt_ilrt_bld_prtest
x64_release_windows_nt_standalone_gc_bld
x64_release_windows_nt_standalone_gc_bld_prtest
x86_arm_altjit_checked_windows_nt_innerloop_bld
x86_arm_altjit_checked_windows_nt_innerloop_bld_prtest
x86_arm_altjit_debug_windows_nt_innerloop_bld
x86_arm_altjit_debug_windows_nt_innerloop_bld_prtest
x86_arm_altjit_release_windows_nt_innerloop_bld
x86_arm_altjit_release_windows_nt_innerloop_bld_prtest
x86_debug_windows_nt_prtest
x86_release_windows_nt_prtest

-- Removed Jobs --

-- Changed Jobs --

```

**Comment by BruceForstall** [comment]
Can you summarize the idea here?

Is the the difference between `innerloop` and `default` that `innerloop` is pri-0, and PR triggered, and `default` is pri-1 and requires a trigger phrase? (Of course, they have different names.)

If `innerloop` is default PR triggered but `default` is not, isn't that confusing (given the word `default`)?

Should `default` be named something like `base` or `baseline` or `normal`?

So innerloop flow jobs depend on innerloop build jobs, and non-innerloop flow jobs depend on non-innerloop build jobs?

File: netci.groovy [hunk 1]
```diff
@@ -2329,13 +2413,13 @@ Constants.allScenarios.each { scenario ->
                     else {
                         // Non-Windows
                         if (architecture == 'arm64') {
-                            if (scenario != 'default' && scenario != 'r2r' && scenario != 'gcstress0x3' && scenario != 'gcstress0xc') {
+                            if (scenario != 'innerloop' && scenario != 'default' && scenario != 'r2r' && scenario != 'gcstress0x3' && scenario != 'gcstress0xc') {
```
Comment (RussKeldorph) [inline netci.groovy]: Small savings: could substitute "!isDefaultOrInnerloop" for "scenario != 'innerloop' && scenarios != 'default'" but honestly this is borderline big enough to warrant a switch table or a table lookup.

**Comment by jashook** [comment]
What you have said is correct. Innerloop named jobs will be PR triggered Pri 0 jobs. Looking at the current list of added jobs it looks like there are non-referenced jobs which need to be removed as everything that is not innerloop pr triggered will now be pri 1.

As for the confusion between default/base/baseline/normal. I have no preferences over names,  default currently  is just the empty name that we use in jitci.

```
So innerloop flow jobs depend on innerloop build jobs, and non-innerloop flow jobs depend on non-innerloop build jobs?
```

Correct

**Comment by RussKeldorph** [comment]
I prefer to use a different term than "default."  I'd probably go with "plain" as it seems the simplest term that maybe isn't overloaded.  Other ideas would be "regular" and "ordinary."  If I wanted to sound fancy, I would use "unqualified."

---
In reply to: [357005465](https://github.com/dotnet/coreclr/pull/15813#issuecomment-357005465) [](ancestors = 357005465)

File: netci.groovy [hunk 2]
```diff
@@ -744,14 +747,16 @@ def static getJobName(def configuration, def architecture, def os, def scenario,
...
+
     def suffix = scenario != 'default' ? "_${scenario}" : '';
     if (isBuildOnly) {
         suffix += '_bld'
     }
     def baseName = ''
     switch (architecture) {
         case 'x64':
-            if (scenario == 'default') {
+            if (isDefaultOrInnerloop) {
```
Comment (BruceForstall) [inline netci.groovy]: For all the new, innerloop jobs, I propose their names should include the architecture in all cases (which specifically means the x64 case), since there is no back compat issue.

File: netci.groovy [hunk 3]
```diff
@@ -813,7 +818,7 @@ def static addNonPRTriggers(def job, def branch, def isPR, def architecture, def
 
     // Check scenario.
     switch (scenario) {
-        case 'default':
+        case ['default', 'innerloop']:
```
Comment (BruceForstall) [inline netci.groovy]: It seems like if we are going to trigger pri-1 (`default`) jobs on push, and those are always a superset of `innerloop`, that we should never have `innerloop` push jobs. Same for cron jobs: no need for any innerloop cron jobs if the equivalent, superset, pri-1 job is there.

Comment (RussKeldorph) [reply netci.groovy]: :+1:

File: netci.groovy (hunk 2 above)
Comment (RussKeldorph) [reply netci.groovy]: 👍

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@BruceForstall I believe this pr is now good to go ptal

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
This pr also addresses #15963

File: netci.groovy [hunk 4]
```diff
@@ -1134,83 +1188,102 @@ def static addTriggers(def job, def branch, def isPR, def architecture, def os,
...
+
                 break
             }
 
             switch (os) {
                 // OpenSUSE, Debian & RedHat get trigger phrases for pri 0 build, and pri 1 build & test
                 case 'Debian8.4':
                 case 'RHEL7.2':
-                    if (scenario == 'default') {
+                    if (isNormalOrInnerloop) {
```
Comment (BruceForstall) [inline netci.groovy]: Won't this mean the same trigger phrase for both `innerloop` and `normal` jobs?

Also, the trigger phrase doesn't include `architecture` or `configuration`. Can't we fix that (namely, make sure all trigger phrases include os/architecture/configuration)? I generally (but not always) avoided changing what was already there, but I think for the new (innerloop) jobs, we should make them more verbose/complete.

Shouldn't the innerloop trigger phrases also include the word "innerloop"?

File: netci.groovy [hunk 5]
```diff
@@ -1134,83 +1188,102 @@ def static addTriggers(def job, def branch, def isPR, def architecture, def os,
...
                         assert !isFlowJob
                         Utilities.addGithubPRTriggerForBranch(job, branch, "${os} ${architecture} ${configuration} Build", "(?i).*test\\W+${os}.*")
                     }
                     break
+
                 case 'Ubuntu16.04':
                     assert !isFlowJob
-                    assert scenario == 'default'
+                    assert isNormalOrInnerloop
```
Comment (BruceForstall) [inline netci.groovy]: same comment about trigger phrases applies here, and probably below, elsewhere, as well.

File: netci.groovy [hunk 6]
```diff
@@ -1500,57 +1613,72 @@ def static addTriggers(def job, def branch, def isPR, def architecture, def os,
...
                     else {
                         Utilities.addGithubPRTriggerForBranch(job, branch, "${os} ${architecture} ${configuration} Build and Test",
                             "(?i).*test\\W+${os}\\W+${architecture}\\W+${configuration}.*")
                     }
                     break
+
+                case 'normal':
+                    if (configuration == 'Checked') {
+                        Utilities.addGithubPRTriggerForBranch(job, branch, "${os} ${architecture} ${configuration} Build and Test",
+                            "(?i).*test\\W+${os}\\W+${architecture}\\W+${configuration}.*")
```
Comment (BruceForstall) [inline netci.groovy]: This trigger phrase looks like a subset of the ones below (e.g., ilrt). So if you trigger an ilrt run, you'll also get this. Seems like the phrase needs to include "Build and Test" at the end.

File: netci.groovy [hunk 7]
```diff
@@ -2183,9 +2308,26 @@ Constants.allScenarios.each { scenario ->
                                     return
                                 }
                                 break
-                            case 'default':
+                            case 'normal':
                                 // Nothing skipped
                                 break
+                            case 'innerloop':
+                                def validOsPrTriggerArchConfigs = Constants.prTriggeredValidInnerLoopCombos[os]
```
Comment (BruceForstall) [inline netci.groovy]: I think you should make this a function, e.g.: `isValidPrTriggeredInnerLoopJob(os, architecture, configuration)`

File: netci.groovy [hunk 8]
```diff
@@ -2198,6 +2340,11 @@ Constants.allScenarios.each { scenario ->
                     switch (architecture) {
                         case 'x86_arm_altjit':
                         case 'x64_arm64_altjit':
+                            // In addition, do not do innerloop testing
+                            if (scenario == "innerloop") {
```
Comment (BruceForstall) [inline netci.groovy]: Doesn't the `Constants.prTriggeredValidInnerLoopCombos` code already cover this (i.e., do you need this?)?

File: netci.groovy [hunk 9]
```diff
@@ -2333,14 +2481,22 @@ Constants.allScenarios.each { scenario ->
                     }
                     else {
                         // Non-Windows
+                        def validLinuxArm64Scenarios = [
```
Comment (BruceForstall) [inline netci.groovy]: I think you should hoist this up next to `validArmWindowsScenarios`, as part of `Constants`

File: netci.groovy [hunk 10]
```diff
@@ -2399,27 +2555,52 @@ Constants.allScenarios.each { scenario ->
...
                                 break
-                            case 'default':
+
+                            case 'normal':
+                                // Nothing skipped
+                                break
+
+                            case 'innerloop':
                                 // Nothing skipped
+                                def validOsPrTriggerArchConfigs = Constants.prTriggeredValidInnerLoopCombos[os]
```
Comment (BruceForstall) [inline netci.groovy]: same comment as above about creating a function for this

File: netci.groovy (hunk 4 above)
Comment (jashook) [reply netci.groovy]: This is a good catch thank you.

**Comment by jashook** [comment]
@dotnet-bot help

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@BruceForstall ptal

**Comment by jashook** [comment]
@dotnet-bot test ci please
